src/MongoDB.Driver/FilterDefinitionBuilder.cs

```diff
@@ -1541,11 +1551,11 @@ private static void AddClause(BsonDocument document, BsonElement clause)
             else if (document.Contains(clause.Name))
             {
                 var existingClause = document.GetElement(clause.Name);
-                if (existingClause.Value is BsonDocument && clause.Value is BsonDocument)
+                if (existingClause.Value is BsonDocument existingClauseValue && clause.Value is BsonDocument clauseValue)
                 {
-                    var clauseValue = (BsonDocument)clause.Value;
-                    var existingClauseValue = (BsonDocument)existingClause.Value;
-                    if (clauseValue.Names.Any(op => existingClauseValue.Contains(op)))
+                    var topLevelElementName = clauseValue.ElementCount > 0 ? clauseValue.GetElement(0).Name : null;
```

Comment:
Please rename this variable:

    clauseOperator

Reply:
Question: how can `clauseValue.ElementCount` ever be 0?

Reply:
it's possible in the case when operator body is empty. See this test: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/130/files#diff-9c6d64d11defdd6e9847744a1776c574R96